core/input/input.cpp

```diff
@@ -697,29 +697,28 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 	for (const KeyValue<StringName, InputMap::Action> &E : InputMap::get_singleton()->get_action_map()) {
 		if (InputMap::get_singleton()->event_is_action(p_event, E.key)) {
 			Action &action = action_state[E.key];
+			bool is_joystick = jm.is_valid();
 			bool is_pressed = false;
-
+			bool is_joystick_valid_zone_enter = false;
 			if (!p_event->is_echo()) {
 				if (p_event->is_action_pressed(E.key)) {
```

Comment:
The variables scope can be reduced.
```suggestion
			if (!p_event->is_echo()) {
				if (p_event->is_action_pressed(E.key)) {
					bool is_joystick_valid_zone_enter = false;
```